src/pages/home/HeaderView.tsx

```diff
@@ -187,9 +191,12 @@ function HeaderView({report, parentReportAction, onNavigationMenuButtonClicked,
 
     // If the onboarding report is directly loaded, shouldShowDiscountBanner can return wrong value as it is not
     // linked to the react lifecycle directly. Wait for trial dates to load, before calculating.
-    // eslint-disable-next-line react-compiler/react-compiler
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    const shouldShowDiscount = useMemo(() => shouldShowDiscountBanner() && !isArchivedReport(reportNameValuePairs), [firstDayFreeTrial, lastDayFreeTrial, reportNameValuePairs]);
+    const shouldShowDiscount = useMemo(
+        () => shouldShowDiscountBanner() && !isArchivedReport(reportNameValuePairs) && hasTeam2025Pricing && subscriptionPlan !== CONST.POLICY.TYPE.TEAM,
+        // eslint-disable-next-line react-compiler/react-compiler
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [firstDayFreeTrial, lastDayFreeTrial, hasTeam2025Pricing, reportNameValuePairs, subscriptionPlan],
+    );
```

Comment:
Do we need to check any of these?

```

    if (!firstDayFreeTrial || !lastDayFreeTrial || !discountInfo) {
        return null;
    }
``

Reply:
i dont think so. they are null checks and these nvps should always be present when the user is on trial

<img width="407" alt="image" src="https://github.com/user-attachments/assets/1ed0326d-76c0-4d0e-aa82-30b2244029c1" />

Reply:
I guess I'm more wondering if we should encapsulate shouldShowDiscount into a utility method so that we don't run into a situation where another check is added within EarlyDiscountBanner that isn't here and we run into the same bug

Reply:
there is a function for it we could move everything there - https://github.com/rushatgabhane/exfy/blob/acaac69ed47ab94c54f80cd60828cc6bc277f357/src/libs/SubscriptionUtils.ts#L287

Reply:
Yeah that seems best, then we can clean up the code a bit

Reply:
That seems like where we should have placed 

```
    if (hasTeam2025Pricing && subscriptionPlan === CONST.POLICY.TYPE.TEAM) {
        return null;
    }

```

actually

Reply:
like the idea

Reply:
> hasTeam2025Pricing && subscriptionPlan

i think it wasn't added in that util because they are derived from hooks.